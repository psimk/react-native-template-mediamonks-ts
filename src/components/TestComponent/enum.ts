export enum LOCALIZED {
  Title = 'gettingStarted.title',
  Description = 'gettingStarted.description',
}

export enum COPY {
  IncrementTitle = 'Increment counter value',
  DecrementTitle = 'Decrement counter value',
  CounterValue = 'Counter value:',
  AnotherCounterValue = 'Another Counter value:',
}
