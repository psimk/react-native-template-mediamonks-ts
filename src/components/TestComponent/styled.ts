import Button from '../Button';
import styled from 'styled-components/native';
import { typography } from '../../themes';

export const Container = styled.View`
  flex: 1;
  margin-top: 50;
  align-items: center;
  margin-horizontal: 20;
`;

export const CounterContainer = styled.View`
  flex: 1;
  justify-content: center;
  align-items: center;
  margin-bottom: 30;
`;

export const CounterButton = styled(Button)`
  margin-vertical: 10;
`;

export const Welcome = styled(typography.H1)`
  text-align:center;
`;

export const Description = styled(typography.P1)`
  text-align:center;
  margin-vertical:15;
`;
