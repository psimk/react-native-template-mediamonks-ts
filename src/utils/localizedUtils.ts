/* eslint-disable import/prefer-default-export */
import I18n from 'react-native-i18n';
import en from '../locales/en';

I18n.fallbacks = true;

I18n.translations = {
	en
};

export const getLocalizedString = (...value) => I18n.t(...value);
