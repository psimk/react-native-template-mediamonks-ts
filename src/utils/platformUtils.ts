/* eslint-disable import/prefer-default-export */
import { Platform } from 'react-native';

export const isIOS = Platform.OS === 'ios';
