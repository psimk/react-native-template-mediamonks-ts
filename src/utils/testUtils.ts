/* eslint-disable import/prefer-default-export */
// import configureStore from 'redux-mock-store';
import thunk from 'redux-thunk';

// create a middleware const that imitates the stores middleware.
const middleware = [thunk];
// Create a mock redux store.
// export const mockStore = configureStore(middleware);
