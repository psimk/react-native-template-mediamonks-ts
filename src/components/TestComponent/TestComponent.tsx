import React from 'react';
import { Text, View } from 'react-native';
import { testComponent } from '../../enum/TestIDs';
import Button from '../Button';
import { getLocalizedString } from '../../utils/localizedUtils';
import styled from 'styled-components';
import { typography } from '../../themes';

const Container = styled.View`
  flex: 1;
  margin-top: 50;
  align-items: center;
  margin-horizontal: 20;
`;

const CounterContainer = styled.View`
  flex: 1;
  justify-content: center;
  align-items: 'center';
  margin-bottom: 30;
`;

const CounterButton = styled(Button)`

  margin-vertical: 10;
`;

const Welcome = styled(typography.H1)`
text-align:center;
`;

const Description = styled(typography.P1)`
text-align:center;
margin-vertical:15;
`;

const enum COPY {
  IncrementTitle = 'Increment counter value',
  DecrementTitle = 'Decrement counter value',
  CounterValue = 'Counter value:',
}

const enum LOCALIZED {
  Title = 'gettingStarted.title',
  Description = 'gettingStarted.description',
}

interface IProps {
  counterValue: number;
  incrementCounterValue: () => {};
  decrementCounterValue: () => {};
}

export default ({ counterValue, incrementCounterValue, decrementCounterValue }: IProps) => (
  <Container testID={testComponent.CONTAINER}>
    <Welcome>{getLocalizedString(LOCALIZED.Title)}</Welcome>
    <Description>{getLocalizedString(LOCALIZED.Description)}</Description>
    <CounterContainer>
      <Text testID={testComponent.COUNTER_VALUE}>{`${COPY.CounterValue} ${counterValue}`}</Text>
      <CounterButton
        testID={testComponent.INCREMENT_BUTTON}
        title={COPY.IncrementTitle}
        onPress={incrementCounterValue}
      />
      <CounterButton
        testID={testComponent.DECREMENT_BUTTON}
        title={COPY.DecrementTitle}
        onPress={decrementCounterValue}
      />
    </CounterContainer>
  </Container>
);
