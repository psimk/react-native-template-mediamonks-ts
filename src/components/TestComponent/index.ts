import { compose } from 'redux';
import { connect } from 'react-redux';
import { incrementCounterValue, decrementCounterValue } from '../../store/actions/counterActions';
import TestComponent from './TestComponent';

const mapStateToProps = state => ({ counterValue: state.counter.counterValue });
const mapDispatchToProps = { incrementCounterValue, decrementCounterValue };

const connected = connect(
  mapStateToProps,
  mapDispatchToProps,
);

export default compose(connected)(TestComponent);
