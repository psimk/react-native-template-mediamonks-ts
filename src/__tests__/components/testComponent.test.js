import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import TestComponent from '../../components/TestComponent';
import { initialState } from '../../store/reducers/counterReducer';
import {
  DECREMENT_COUNTER_VALUE,
  INCREMENT_COUNTER_VALUE,
} from '../../store/actions/counterActions';
import { mockStore } from '../../utils/testUtils';

// Add the initial state to our mock redux store.
const counter = { ...initialState };

/**
 *  Test Component
 */
describe('Test component container', () => {
  let store;
  let container;

  /**
   *  Before each create a reference of the test component with a mock store.
   */
  beforeEach(() => {
    store = mockStore({ counter });
    store.dispatch = jest.fn();
    container = shallow(<TestComponent store={store} />);
  });

  it('Counter: renders correctly', () => {
    const containerJson = renderer.create(container).toJSON();
    expect(containerJson).toMatchSnapshot();
  });
  /**
   * Components initial props
   */
  it('Check mapState and dispatch to props of component', () => {
    expect(container.props()).toEqual(
      expect.objectContaining({
        counterValue: 0,
        increment: expect.any(Function),
        decrement: expect.any(Function),
      }),
    );
  });
  /**
   *  Component prop dispatches
   */
  it('Test dispatch of increment action', () => {
    container.props().increment();
    expect(store.dispatch).toHaveBeenCalledWith({ type: INCREMENT_COUNTER_VALUE });
  });
  it('Test dispatch of decrement action', () => {
    container.props().decrement();
    expect(store.dispatch).toHaveBeenCalledWith({ type: DECREMENT_COUNTER_VALUE });
  });
});
