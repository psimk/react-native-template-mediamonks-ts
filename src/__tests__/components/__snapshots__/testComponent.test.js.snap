// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test component container Counter: renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "marginHorizontal": 20,
      "marginTop": 50,
    }
  }
  testID="test-component-container"
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "color": "black",
        "fontFamily": "Helvetica Neue",
        "fontSize": 26,
        "textAlign": "center",
      }
    }
  >
    Welcome to the 
 React-Native Mediamonks template project
  </Text>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "color": "black",
        "fontFamily": "Helvetica Neue",
        "fontSize": 16,
        "marginVertical": 15,
        "textAlign": "center",
      }
    }
  >
    This is a basic example of a project setup. Below is a simple counter to show how redux is setup.
  </Text>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
        "marginBottom": 30,
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      testID="test-component-counter-value"
    >
      Counter value: 
      0
    </Text>
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "red",
          "borderRadius": 10,
          "justifyContent": "center",
          "marginVertical": 10,
          "opacity": 1,
          "padding": 20,
        }
      }
      testID="test-component-button-increment"
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "white",
            },
            undefined,
          ]
        }
      >
         
        Increment counter value
         
      </Text>
    </View>
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "red",
          "borderRadius": 10,
          "justifyContent": "center",
          "marginVertical": 10,
          "opacity": 1,
          "padding": 20,
        }
      }
      testID="test-component-button-decrement"
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "white",
            },
            undefined,
          ]
        }
      >
         
        Decrement counter value
         
      </Text>
    </View>
  </View>
</View>
`;
